packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts

```diff
@@ -265,6 +272,14 @@ export class RelationalDatabaseConnectionValueState extends ConnectionValueState
         dataSpec = new SpannerDatasourceSpecification('', '', '', '', '');
         break;
       }
+      case CORE_DATASOURCE_SPEC_TYPE.TRINO: {
+        dataSpec = new TrinoDatasourceSpecification(
+          '',
+          NaN,
```

Comment:
why do we use NaN here?

Reply:
This filed is of number type and we dont want to give any default value, thats why NaN. Is there a better way to achieve the same ?

Reply:
I personally don't prefer NaN. Maybe we could give a dummy default value between 1024 and 65535

Reply:
or maybe undefined?